main.go

```diff
@@ -247,6 +247,19 @@ func (p *prometheus) reloadConfig() bool {
 // down. The method installs an interrupt handler, allowing to trigger a
 // shutdown by sending SIGTERM to the process.
 func (p *prometheus) Serve() {
+	// Wait for reload or termination signals. Start the handler for SIGHUP as
+	// early as possible, but ignore it until we are ready to handle reloading
+	// our config.
+	hup := make(chan os.Signal)
+	block := make(chan bool)
```

Comment:
Could you call this `hupReady` or something that makes it easier to understand the purpose?